api/ask_astro/services/questions.py

```diff
@@ -27,6 +28,20 @@ async def _update_firestore_request(request: AskAstroRequest) -> None:
     )
 
 
+def _preprocess_request(request: AskAstroRequest) -> None:
+    if len(request.prompt) > 20000:
+        error_msg = "Question text is too long. Please try making a new thread and shortening your question."
+        request.response = error_msg
+        raise Exception(error_msg)
+    if len(request.prompt) == 0:
+        error_msg = "Question text cannot be empty. Please try again with a different question."
+        request.response = error_msg
+        raise Exception(error_msg)
+    if len(request.messages) > 10:
+        request.messages = request.messages[-10:]
```

Comment:
why -10?

Reply:
Should this be a parameter instead so that it can be changed as part of env in deployment?

Reply:
Fixed!